homeassistant/components/device_tracker/mqtt_json.py

```diff
@@ -0,0 +1,88 @@
+"""
+Support for GPS tracking MQTT enabled devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.mqtt_json/
+"""
+import asyncio
+import json
+import logging
+
+import voluptuous as vol
+
+import homeassistant.components.mqtt as mqtt
+from homeassistant.core import callback
+from homeassistant.const import CONF_DEVICES
+from homeassistant.components.mqtt import CONF_QOS
+from homeassistant.components.device_tracker import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['mqtt']
+
+_LOGGER = logging.getLogger(__name__)
+
+LAT_KEY = 'lat'
+LON_KEY = 'lon'
+ACCURACY_KEY = 'acc'
+BATTERY_KEY = 'batt'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(mqtt.SCHEMA_BASE).extend({
+    vol.Required(CONF_DEVICES): {cv.string: mqtt.valid_subscribe_topic},
+})
+
+
+@asyncio.coroutine
+def async_setup_scanner(hass, config, async_see, discovery_info=None):
+    """Setup the MQTT tracker."""
+    devices = config[CONF_DEVICES]
+    qos = config[CONF_QOS]
+
+    dev_id_lookup = {}
+
+    @callback
+    def async_tracker_message_received(topic, payload, qos):
+        """MQTT message received."""
+        dev_id = dev_id_lookup[topic]
+
+        try:
+            data = json.loads(payload)
+        except ValueError:
+            _LOGGER.error('Unable to parse payload as JSON: %s', payload)
+            return
+
+        if not isinstance(data, dict):
```

Comment:
Could please point me to an example? I couldn't find any other tracker component validating passed data with voluptuous. Thanks